pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -383,11 +394,12 @@ public void sendAsync(Message<?> message, SendCallback callback) {
 
         // If compression is enabled, we are compressing, otherwise it will simply use the same buffer
         ByteBuf compressedPayload = payload;
+        boolean compressed = false;
         // Batch will be compressed when closed
         // If a message has a delayed delivery time, we'll always send it individually
         if (!isBatchMessagingEnabled() || msgMetadata.hasDeliverAtTime()) {
```

Comment:
@eolivelli Is change to `!isBatchMessagingEnabled() || msgMetadata.hasDeliverAtTime() || isSchemaReady` works? It's not a big problem, I just noticed the `compressed` passed to the following methods, if we can make the compression works here for the multiple schema producer, I think we can void pass the `compressed`.

The change looks good to me.

Reply:
I would prefer to keep the code simpler.
it is already quite complicated.

thanks @codelipenghui  for your review!